castai/resource_node_template.go

```diff
@@ -223,6 +223,7 @@ func resourceNodeTemplate() *schema.Resource {
 							Optional:    true,
 							Default:     false,
 							Description: "Storage optimized instance constraint - will only pick storage optimized nodes if true",
+							Deprecated:  "Storage optimized constraint will be removed in the next major version and other resource for this functionality will have to be used.",
```

Comment:
This wording doesn't sound right, I feel it gives the impression we are going to be removing the constraint completely. Also we wont be replacing this with a different resource, just a different argument

Reply:
Hmm, agree, what do you think about this:
`Storage optimized constraint is deprecated and will be removed in the next major version. Other argument for the same functionality will be provided with the major version.`
Maybe you can think of other variations?

Reply:
`Storage optimized constraint is deprecated and will be replaced by a new argument in the next major release`?

Reply:
updated